testing/MLDBFB-440_error_on_ds_wo_cols.py

```diff
@@ -0,0 +1,24 @@
+#
+# MLDBFB-440_error_on_ds_wo_cols.py
+# Mich, 2016-03-30
+# Copyright (c) 2016 Datacratic Inc. All rights reserved.
+#
+
+import unittest
+
+mldb = mldb_wrapper.wrap(mldb)  # noqa
+
+class TestErrorOnDsWoCols(MldbUnitTest):  # noqa
+
+    @unittest.expectedFailure
+    def test_it(self):
+        ds = mldb.create_dataset({'id' : 'noColDs', 'type' : 'sparse.mutable'})
+        ds.record_row('row1', [])
```

Comment:
My rationale is: I can apply a transform over a dataset in such a way that some of the rows no longer have columns.
Hence, we should be able to create empty rows directly.